lib/src/eval/compiler/expression/prefix.dart

```diff
@@ -34,8 +37,44 @@ Variable compilePrefixExpression(CompilerContext ctx, PrefixExpression e) {
   if (method == "!") {
     return V.invoke(ctx, method, []).result;
   }
-  final zero = V.type == CoreTypes.int.ref(ctx)
-      ? BuiltinValue(intval: 0)
-      : BuiltinValue(doubleval: 0.0);
-  return zero.push(ctx).invoke(ctx, method, [V]).result;
+
+  if ([TokenType.PLUS_PLUS, TokenType.MINUS_MINUS].contains(e.operator.type)) {
+    return _handleDoubleOperands(e, ctx);
+  }
+
+  return _zeroForType(V.type, ctx).push(ctx).invoke(ctx, method, [V]).result;
+}
+
+BuiltinValue _zeroForType(TypeRef type, CompilerContext ctx) =>
+    type == CoreTypes.int.ref(ctx)
+        ? BuiltinValue(intval: 0)
+        : BuiltinValue(doubleval: 0.0);
+
+BuiltinValue _oneForType(TypeRef type, CompilerContext ctx) =>
+    type == CoreTypes.int.ref(ctx)
+        ? BuiltinValue(intval: 1)
+        : BuiltinValue(doubleval: 1.0);
+
+Variable _handleDoubleOperands(PrefixExpression e, CompilerContext ctx) {
+  var V = compileExpression(e.operand, ctx);
+  final ref = compileExpressionAsReference(e.operand, ctx);
```

Comment:
Compiling the same expression (`e.operand` here) twice is not allowed. Should be pretty easy to get rid of V and just save whatever data about L you need here

Reply:
I'm now realizing you're actually compiling it three times because the first line of compilePrefixExpression also compiles it. Any given expression should definitely only be compiled once so you'll have to change some things around